app/src/main/java/com/yacgroup/yacguide/TourbookActivity.kt

```diff
@@ -61,7 +61,7 @@ class TourbookActivity : BaseNavigationActivity() {
     private lateinit var _db: DatabaseWrapper
     private lateinit var _customSettings: SharedPreferences
     private lateinit var _availableYears: IntArray
-    private lateinit var _tourbookExporter: TourbookExporter
+    private var _tourbookExportFormat: TourbookExportFormat = TourbookExportFormat.eJSON
```

Comment:
Actually I thought about this, too but why should I habe this class member, if I just use it inside the export method?

Reply:
You also use it `selectExportFormat()` since you assign the concrete exporter there.

Reply:
The problem with this solution is that you now need both, the exporter and the format, as class properties because the format is used for selecting the file extension and for translation of the selected item in the UI and the actual format.

Reply:
Really? Wouldn't this work [pseudo]? 
```
_selectExportFormat() {
    val format = getFromUI() 
    _exporter = Factory.create(format) 
    selectExportFile(format) 
} 
export() {
    _exporter.export() 
}

Reply:
You are right, it is possible but than we definitely loose the feature that the last choice for the format is remembered.

Reply:
Ok but why should somebody export the tourbook twice within the same activity session? I mean, once you leave `TourbookActivity` `_tourbookExportFormat` is reset to json anyway if you re-enter.

Reply:
I agree.